### PR TITLE
refactored copy_ssh_key() to handle already existing priv rsa key

### PR DESCRIPTION
**Pull request by lpramuk**

### CODE REVIEW

File: automation_tools/satellite6/upgrade/tools.py [hunk 1]
```diff
@@ -37,20 +37,25 @@ def copy_ssh_key(from_host, to_host):
...
-        'ssh-keygen -f ~/.ssh/id_rsa -t rsa -N \'\'', warn_only=True),
-        host=from_host)
-    if int(execute(lambda: run('[ -f ~/.ssh/id_rsa.pub ]; echo $?'),
-                   host=from_host)[from_host]) == 0:
-        pub_key = execute(lambda: run(
-            'cat ~/.ssh/id_rsa.pub'), host=from_host)[from_host]
-        execute(lambda: run('[ ! -f ~/.ssh/authorized_keys ] && '
-                            'touch ~/.ssh/authorized_keys',
-                            warn_only=True), host=to_host)
+        '[ ! -f ~/.ssh/id_rsa ] || '
```
Comment (elyezer) [inline automation_tools/satellite6/upgrade/tools.py]: ! -f means that the file does not exist, so accessing it will fail later

Comment (lpramuk) [inline automation_tools/satellite6/upgrade/tools.py]: Short answer:
It shold read: It doesnt exist OR access it

Long answer:
! -f || cmd is equivalent to -f && cmd
The latter form return 1 as file doesnot exist and failing whole run() while the first form return 0 and run() is ok

Comment (elyezer) [inline automation_tools/satellite6/upgrade/tools.py]: What I don't get is that why looking for `~/.ssh/id_rsa` instead of `~/.ssh/id_rsa.pub`

Comment (lpramuk) [inline automation_tools/satellite6/upgrade/tools.py]: I'm looking for privkey to generate pubkey out of it 
if pubkey is present it is over written resulting in matching keypair (privkey could be from differrent keypait than pubkey)

Comment (lpramuk) [inline automation_tools/satellite6/upgrade/tools.py]: Another approach would be to delete existing privkey and always generate keypair. 

But if I have image/instance with some important privkey I would delete it, so I chose approach to actually re-use existing privkey for our purpose...

Comment (elyezer) [inline automation_tools/satellite6/upgrade/tools.py]: Fair enough

**Comment by elyezer** [comment]
ACK, even though reading the `not or` logic is harder `and` the PR looks good.

**Comment by jyejare** [comment]
ACK

**Pull request merged**